File: backend/app/models/orientation.py
from typing import Literal

from pydantic import BaseModel, EmailStr

from app.models.formation import FormationOut

NiveauOrientation = Literal["debutant", "intermediaire", "avance"]
ModeOrientation = Literal["presentiel", "en_ligne", "peu_importe"]


class OrientationReponses(BaseModel):
    objectif: str
    filiere_cible: str
    niveau: NiveauOrientation
    mode: ModeOrientation
    age: int | None = None


class RecommandationOut(BaseModel):
    formation_principale: FormationOut
    alternatives: list[FormationOut]
    justification: str
    source: Literal["ia", "regles"]


class EnvoyerResultatIn(BaseModel):
    email: EmailStr
    formation_principale_id: str
    alternatives_ids: list[str] = []
    justification: str

File: frontend/src/types/orientation.ts
import type { Formation } from './formation'

export type Objectif = 'parent' | 'jeune_adulte' | 'professionnel' | 'candidat_emigration'
export type NiveauOrientation = 'debutant' | 'intermediaire' | 'avance'
export type ModeOrientation = 'presentiel' | 'en_ligne' | 'peu_importe'

export interface OrientationReponses {
  objectif: Objectif
  filiere_cible: string
  niveau: NiveauOrientation
  mode: ModeOrientation
  age: number | null
}

export interface Recommandation {
  formation_principale: Formation
  alternatives: Formation[]
  justification: string
  source: 'ia' | 'regles'
}
